### ISSUE TITLE
examples/tetris: X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)

### ISSUE THREAD

**Issue by ithgub** [vlang/v#11453]
**V version:** V 0.2.4 c8d4a64.ab3adf3
**OS:** linux, Debian GNU/Linux 10 (buster) — actually, **antiX 19** — systemd-free Debian

https://github.com/vlang/v/blob/master/README.md#testing-and-running-the-examples
`v run tetris/tetris.v`

At first I didn't take into account the notice:
> NB: In order to build Tetris or 2048 (or anything else using sokol or gg graphics modules) on some Linux systems, you need to install libxi-dev and libxcursor-dev .

since it is under code snippets and screenshot. It would be clearer was it above or by side.
Output:
```
> v run tetris/tetris.v
/home/x/.bin/v/thirdparty/stb_image/stbi.o not found, building it in /home/olve/.vmodules/cache/ce/ce68723175c924c16145d9747c1a1ebb.o ...                                           
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)                                                                                                          
  Major opcode of failed request:  155 (NV-GLX)                                            
  Minor opcode of failed request:  4 ()                                                    
  Resource id in failed request:  0xd40008
  Serial number of failed request:  78  
  Current serial number in output stream:  78
``` 
Then those 2 dependencies:
```
> sudo apt install libxi-dev libxcursor-dev
libxcursor-dev is already the newest version (1:1.1.15-2).
libxcursor-dev set to manually installed.
libxi-dev is already the newest version (2:1.7.9-1).
libxi-dev set to manually installed.
```
But again:
```
> v run tetris/tetris.v
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
  Major opcode of failed request:  155 (NV-GLX)
  Minor opcode of failed request:  4 ()
  Resource id in failed request:  0xd40008
  Serial number of failed request:  78
  Current serial number in output stream:  78
```
Next, I found `v/examples/tetris/README.md` which lists 4 instead of 2 dependencies.
`libgl-dev` is `libgl1-mesa-dev` on Debian 10 Buster according to https://pkgs.org/download/libgl-dev
```
> sudo apt update
> sudo apt install libx11-dev libxi-dev libxcursor-dev libgl1-mesa-dev
libx11-dev is already the newest version (2:1.6.7-1+deb10u2).
libxcursor-dev is already the newest version (1:1.1.15-2).
libxi-dev is already the newest version (2:1.7.9-1).
libgl1-mesa-dev is already the newest version (18.3.6-2+deb10u1).
```
Still:
```
> v run tetris/tetris.v
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
  Major opcode of failed request:  155 (NV-GLX)
  Minor opcode of failed request:  4 ()
  Resource id in failed request:  0xd40008
  Serial number of failed request:  78
  Current serial number in output stream:  78
```

**Comment by Larpon** [comment]
Can you run other hardware accelerated software on your system without errors?
Does `glxgears` run for example?

**Comment by ithgub** [comment]
```
> glxgears
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  155 (NV-GLX)
  Minor opcode of failed request:  4 ()
  Resource id in failed request:  0xd40002
  Serial number of failed request:  38
  Current serial number in output stream:  38
```

Recently, after an upgrade (and `xserver` was among upgrades), `Qt`-applications `anki` and `qutebrowser` stopped working, and I couldn't find support so I still have them broken.

I assume that the problem is not in `v` but in my machine so the issue may be closed. Sorry and thank you. But do you happen to know how to fix my stuff?

**Comment by JalonSolov** [comment]
You may need to update your graphics driver, and make sure X is using it.  The driver must support OpenGL 3.3+.

**Comment by ithgub** [comment]
@JalonSolov on the right track.

In parallel, I found [How to fix "X Error of failed request: BadWindow (invalid Window parameter)" ?](https://groups.google.com/g/alt.os.linux/c/N7IUgXKY50E)
> I find that after every kernel upgrade I need to shut down X and re-run
the nvidia setup script
CC=/usr/bin/gcc-4.3 ./nvidia-installer.sh
or something like that. 

I reinstalled nvidia driver (can't update — legacy hardware) and everything works fine.  
I had thought that automated upgrades do upgrades automatically. Naive of me :laughing:
